examples/with-apollo/pages/_app.js

```diff
@@ -0,0 +1,17 @@
+import App, {Container} from 'next/app'
+import React from 'react'
+import withApolloClient from '../lib/with-apollo-client'
+import { ApolloProvider } from 'react-apollo'
+
+class MyApp extends App {
+  render () {
+    const {Component, pageProps, apolloClient} = this.props
+    return <Container>
+      <ApolloProvider client={apolloClient}>
```

Comment:
We have to do this since `Container` writes to context too. Imho this is not a problem and even a bit more explicit.

Reply:
@timneutkens could you document (maybe in a comment), why this is true? Also, do you mean that any HOC will need to inject Providers this way?